ExploreASL_Import.m

```diff
@@ -572,12 +572,23 @@ function ExploreASL_Import(imPar, bCopySingleDicoms, bUseDCMTK, bCheckPermission
                             first_match = xASL_adm_GetFileList(scanpath, ['.*' imPar.dcmExtFilter],'FPList',[0 Inf]);
                             if  ~isempty(first_match); first_match = first_match{1}; end
                         end
-                    end
+					end
 
-                    %% Merge NIfTIs if there are multiples
-					if length(nii_files)>1
-						1;
+					%% In case of a single NII ASL file loaded from PAR/REC, we need to shuffle the dynamics from CCCC...LLLL order to CLCLCLCL... order
```

Comment:
This looks good, beware of the space ` ` after each comma: `, ` instead of `,`

`scanExtension` would capture `.PAR` or `.REC` here, so you want to compare with `'^\.(par|rec)$'` instead of `'par|rec'` . Also, you were missing some brackets here

Reply:
No need to - this works. It returns the index with occurences - no need for brackets and stuff...
Check regexpi('.rec','par|rec')

Reply:
Yes, it works, but this would also work with an extension of `.par.zip` or an extension of `par` (without extension period) which would in both cases crash the script. Here you want to be explicit, we don't want the code to work with occurrences that are different than `.PAR` or `.REC` right?

Reply:
Also, you omitted the comma comment (first line of my reaction here :)

Reply:
Regexp fixed with your comment.

Sorry - added comma - we want the comma, not to remove it :)